kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/application/StorageAwareApplicationCommandRegistry.kt

```diff
@@ -0,0 +1,135 @@
+@file:Suppress(
+    "UNCHECKED_CAST"
+)
+@file:OptIn(
+    ExperimentalCoroutinesApi::class
+)
+
+package com.kotlindiscord.kord.extensions.commands.application
+
+import com.kotlindiscord.kord.extensions.commands.application.message.MessageCommand
+import com.kotlindiscord.kord.extensions.commands.application.slash.SlashCommand
+import com.kotlindiscord.kord.extensions.commands.application.user.UserCommand
+import com.kotlindiscord.kord.extensions.registry.RegistryStorage
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.event.interaction.ChatInputCommandInteractionCreateEvent
+import dev.kord.core.event.interaction.MessageCommandInteractionCreateEvent
+import dev.kord.core.event.interaction.UserCommandInteractionCreateEvent
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.toList
+
+/**
+ * ApplicationCommandRegistry which acts based off a specified storage interface.
+ *
+ * Discord lifecycles may not be implemented in this class and require manual updating.
```

Comment:
Do you mean that this type may not handle syncing commands to Discord? That really should be something that's easy to do for users, so I don't really know whether that's ideal for this class.

Reply:
It only does partial updates, i.e. only registers new, but doesnt check for updates nor deletions.

Reply:
If that's the case, then we really ought to have an advanced tutorial that explains how to do this in a nice way.

Reply:
That will probably make sense, though if someone really needs this, they'll probably know what they're doing already... I hope?

Reply:
I guess so, though there's no reason we can't put something on the wiki later.